UnitySDK/Assets/ML-Agents/Scripts/Utilities.cs

```diff
@@ -28,31 +28,67 @@ public static class Utilities
             var numTextures = textures.Count;
             var width = textures[0].width;
             var height = textures[0].height;
-            var data = tensorProxy.data;
 
             for (var t = 0; t < numTextures; t++)
             {
-                var texturePixels = textures[t].GetPixels32();
-                for (var h = height - 1; h >= 0; h--)
+                var texture = textures[t];
+                if (width != texture.width || height != texture.height)
                 {
-                    for (var w = 0; w < width; w++)
+                    // TODO throw? We didn't do anything about this before.
+                }
+
+                TextureToTensorProxy(texture, tensorProxy, grayScale, t);
+            }
+        }
+
+        /// <summary>
+        /// Puts a Texture2D into a TensorProxy.
+        /// </summary>
+        /// <param name="texture">
+        /// The texture to be put into the tensor.
+        /// </param>
+        /// <param name="tensorProxy">
+        /// TensorProxy to fill with Texture data.
+        /// </param>
+        /// <param name="grayScale">
+        /// If set to <c>true</c> the textures will be converted to grayscale before
+        /// being stored in the tensor.
+        /// </param>
+        /// <param name="textureOffset">
+        /// Index of the texture being written.
+        /// </param>
+        public static void TextureToTensorProxy(
+            Texture2D texture,
+            TensorProxy tensorProxy,
+            bool grayScale,
+            int textureOffset = 0)
+        {
+            var width = texture.width;
+            var height = texture.height;
+            var data = tensorProxy.data;
+
+            var t = textureOffset;
+            var texturePixels = texture.GetPixels32();
+            for (var h = height - 1; h >= 0; h--)
```

Comment:
can we add a comment here about `h--`

Reply:
Yeah, this is the old code but probably unindented one level. Need to merge that other PR and update this.

Reply:
Will do this after merge